src/Razor/src/Microsoft.VisualStudio.LanguageServices.Razor/RoslynProjectChangeDetector.Comparer.cs

```diff
@@ -8,7 +8,7 @@
 
 namespace Microsoft.VisualStudio.Razor;
 
-internal partial class WorkspaceProjectStateChangeDetector
+internal partial class RoslynProjectChangeDetector
 {
     private sealed class Comparer : IEqualityComparer<(Project?, IProjectSnapshot)>
```

Comment:
💭 I just realized that this comparer breaks the contract of `ProjectKey` as an opaque key somewhat by using `FilePathComparer` for equality and hash code. However, `ProjectKey` itself internally uses `FilePathComparer` and also correctly handles `ProjectKey.Unknown`. Should this be updated to just do this?

```C#
        public bool Equals((Project?, IProjectSnapshot) x, (Project?, IProjectSnapshot) y)
        {
            var (_, snapshotX) = x;
            var (_, snapshotY) = y;

            return snapshotX.Key == snapshotY.Key;
        }
        public int GetHashCode((Project?, IProjectSnapshot) obj)
        {
            var (_, snapshot) = obj;
            return snapshot.Key.GetHashCode();
        }
```